consensus/istanbul/backend/handler.go

```diff
@@ -117,6 +118,17 @@ func (sb *Backend) NewChainHead() error {
 		} else {
 			sb.logger.Info("Validators Election Results: Node IN ValidatorSet")
 		}
+
+		// Set the randomness for proposer selection ordering from the block.
+		seed, err := random.Random(currentBlock.Header(), nil)
```

Comment:
We should always pass a header and statedb to these functions, *particularly* when it would affect consistency

Reply:
Added a call to stateAt to explicitly get the state